### PR TITLE
Add competence profile data model

### PR DESCRIPTION
**Pull request by HansenSven**

### CODE REVIEW

**Review by NealGeilen — APPROVED** [review]

File: Epsilon.Host.WebApi/Controllers/ComponentController.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        );
+    }
+    
+    private static T GetRandom<T>(IEnumerable<T> items)
+    {
+        var random = new Random();
+        var itemsArray = items.ToArray();
+        var index = random.Next(0, itemsArray.Length);
+        return itemsArray.ElementAt(index);
+    }
```
Comment (Typiqally) [inline Epsilon.Host.WebApi/Controllers/ComponentController.cs:64]: Remove

File: Epsilon.Canvas.Abstractions/QueryResponse/GetUserSubmissionOutcomes.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+﻿using System.Text.Json.Serialization;
+
+namespace Epsilon.Canvas.Abstractions.QueryResponse;
+
+public record GetUserSubmissionOutcomes(
```
Comment (Typiqally) [inline Epsilon.Canvas.Abstractions/QueryResponse/GetUserSubmissionOutcomes.cs:6]: Move these to separate records in GraphQL namespace in the Canvas project

File: Epsilon.Canvas.Abstractions/ICanvasModuleCollectionFetcher.cs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿using Epsilon.Canvas.Abstractions.Model;
+using Epsilon.Canvas.Abstractions.QueryResponse;
```
Comment (Typiqally) [inline Epsilon.Canvas.Abstractions/ICanvasModuleCollectionFetcher.cs]: Remove

**Review by Typiqally — CHANGES_REQUESTED** [review]

File: Epsilon.Host.WebApi/Controllers/ComponentController.cs (hunk 1 above)
Comment (HansenSven) [reply Epsilon.Host.WebApi/Controllers/ComponentController.cs:64]: We both looked over it this afternoon... it's being used in the for loop above to create a mock competence profile

**Review by Typiqally — APPROVED** [review]

**Pull request merged**